pkg/operator/ceph/cluster/mgr/dashboard.go

```diff
@@ -148,30 +150,30 @@ func (c *Cluster) configureDashboardModuleSettings(daemonID string) (bool, error
 	return hasChanged, nil
 }
 
-func (c *Cluster) initializeSecureDashboard() (bool, error) {
+func (c *Cluster) initializeSecureDashboard() error {
 	// we need to wait a short period after enabling the module before we can call the `ceph dashboard` commands.
 	time.Sleep(dashboardInitWaitTime)
 
 	password, err := c.getOrGenerateDashboardPassword()
 	if err != nil {
-		return false, errors.Wrap(err, "failed to generate a password for the ceph dashboard")
+		return errors.Wrap(err, "failed to generate a password for the ceph dashboard")
 	}
 
 	if c.spec.Dashboard.SSL {
 		alreadyCreated, err := c.createSelfSignedCert()
 		if err != nil {
-			return false, errors.Wrap(err, "failed to create a self signed cert for the ceph dashboard")
+			return errors.Wrap(err, "failed to create a self signed cert for the ceph dashboard")
 		}
 		if alreadyCreated {
-			return false, nil
+			return nil
 		}
```

Comment:
How about restarting the dashboard after generating the cert the first time, just to be safe?
```suggestion
		} else {
			if err := c.restartDashboard(); err != nil {
				logger.Warningf("failed to restart dashboard after generating ssl cert. %v", err)
			}
		}
```